aea/decision_maker/scaffold.py

```diff
@@ -0,0 +1,189 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains a scaffold of the decision maker class and auxilliary classes."""
+
+from typing import List
+
+from aea.decision_maker.base import DecisionMaker as BaseDecisionMaker
+from aea.decision_maker.base import LedgerStateProxy as BaseLedgerStateProxy
+from aea.decision_maker.base import OwnershipState as BaseOwnershipState
+from aea.decision_maker.base import Preferences as BasePreferences
+from aea.decision_maker.messages.base import InternalMessage
+from aea.decision_maker.messages.transaction import TransactionMessage
+
+
```

Comment:
We can let the user scaffold a dm as a single python file in the root of the project. Scaffold will copy the file there and then add the filepath in aea config for loading to `.active`

Reply:
re. using `.active` please see above. In general, not sure I like the fact that the user should modify modules by hand. We should expose an API that let him doing that at runtime (maybe declaring it in the `aea-config.yaml` so the framework knows which class to replace)

Reply:
Also, what about using the `Strategy` pattern? The `DecisionMaker` class remains the same, whereas the user can control the decision maker behaviour via composition:

https://en.wikipedia.org/wiki/Strategy_pattern

Reply:
Followed your advice and went for strategy pattern. We now create a handler for the decision maker, similar to the handler in the skills. The difference is, that objects passed to `kwargs` will be exposed to each skill context, whilst identity and crucially wallet are only accessible to dm handler. the ledger_apis will eventually be removed here.